### PR TITLE
CI/TST: Disallow usage of pytest.xfail

### LINKED ISSUE #38902: TST: Disallow pytest.xfail
pytest.xfail will immediately stop a test, not checking if it would have otherwise succeeded. The request fixture can used to run a test and only pass if the test does indeed fail:

    request.node.add_marker(pytest.mark.xfail(reason="..."))

 - [x] Remove all occurrences of `pytest.xfail`
 - [ ] Update docs
 - [ ] Add pre-commit hook

PR #38901 handles many of the current occurrences. The remaining ones are in:

 - [x] indexes/test_common
 - [x] pandas/conftest
 - [x] indexes/test_setops
 - [x] indexing/test_coercion
 - [x] series/indexing/test_setitem
 - [x] series/test_arithmetic
 - [x] series/indexing/test_where
 - [x] tseries/offsets

### PR DESCRIPTION
**Pull request by rhshadrach**
- [x] closes #38902
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -127,6 +127,11 @@ repos:
         types: [python]
         files: ^pandas/tests/
         exclude: ^pandas/tests/extension/
+    -   id: unwanted-patters-pytest-xfail
+        name: Check for use of pytest.xfail
+        entry: pytest\.xfail
+        language: pygrep
+        types: [python]
```
Comment (MarcoGorelli) [inline .pre-commit-config.yaml:134]: Nice! Maybe add `files: ^pandas/tests`?

Comment (rhshadrach) [reply .pre-commit-config.yaml:134]: Added, thanks!

**Comment by jreback** [comment]
looks fine, some doc issue in ci / checks. ping on green.

**Comment by rhshadrach** [comment]
@jreback green

**Comment by jreback** [comment]
thanks @rhshadrach

**Pull request merged**